src/components/Dashboard/RewardsModule.js

```diff
@@ -15,46 +15,74 @@ import NoRewards from './NoRewards'
-    : bigNum(0)
+const useTotalFeeRewards = (arbitrableFees, appealFees, subscriptionFees) => {
+  return useMemo(() => {
+    const totalArbitrableFees = arbitrableFees
+      ? arbitrableFees.reduce((acc, fee) => acc.add(fee.amount), bigNum(0))
+      : bigNum(0)
+
+    const totalAppealFees = appealFees
+      ? appealFees.reduce((acc, fee) => acc.add(fee.amount), bigNum(0))
+      : bigNum(0)
 
-  const totalAppealFees = appealFees
-    ? appealFees.reduce((acc, fee) => acc.add(fee.amount), bigNum(0))
-    : bigNum(0)
+    const totalSubscriptionFees = subscriptionFees
+      ? subscriptionFees.reduce((acc, fee) => acc.add(fee.amount), bigNum(0))
+      : bigNum(0)
 
-  return { totalArbitrableFees, totalAppealFees }
+    return { totalArbitrableFees, totalAppealFees, totalSubscriptionFees }
+  }, [appealFees, arbitrableFees, subscriptionFees])
 }
 
-// Ruling Fees => ANJ fees
-// Dispute fees + Appeal Fees => DAI fees
+// anjRewards => ANJ => Settled onSettleReward
+// feeRewards => DAI => settled onSettleRewards and onSettleAppealDeposit
+// subscriptions fees => DAI => claimed directly from the CourtSubscription contract
+// Once rewards are settled, we must withdraw them from the treasury (onWithdraw)
+// As opposed to fee rewards, subscription fees are directly sent to the juror's wallet after claimed
 const RewardsModule = React.memo(function RewardsModule({
   rewards,
   treasury,
   loading,
-  onWithdraw,
-  onSettleReward,
+  onClaimSubscriptionFees,
   onSettleAppealDeposit,
+  onSettleReward,
+  onWithdraw,
 }) {
   const wallet = useWallet()
   const { feeToken } = useCourtConfig()
 
-  const { totalArbitrableFees, totalAppealFees } = useTotalDisputesFees(
-    rewards?.arbitrableFees,
-    rewards?.appealFees
+  // Subscriptions are fetched directly from the subscriptions contract
+  const [subscriptionFees, setSubscriptionFees] = useJurorSubscriptionFees()
+  const { anjRewards, feeRewards } = rewards || {}
+
+  const {
+    totalArbitrableFees,
+    totalAppealFees,
+    totalSubscriptionFees,
+  } = useTotalFeeRewards(
+    feeRewards?.arbitrableFees,
+    feeRewards?.appealFees,
+    subscriptionFees
   )
 
+  // We'll get the total juror's balance held in the treasury
   // TODO: Handle possible multiple tokens (fee token can change)
   const treasuryToken = treasury?.find(({ token }) =>
     addressesEqual(token.id, feeToken.id)
   )
   const treasuryBalance = treasuryToken ? treasuryToken.balance : bigNum(0)
+
+  // Total dispute fees are conformed by appeal fees and arbitrable fees (fees paid by the creator of the dispute for the first round
+  // and fees paid by appealers for subsequent rounds)
   const totalDisputesFees = totalArbitrableFees.add(totalAppealFees)
-  const totalFees = totalDisputesFees.add(treasuryBalance)
+
+  // All dispute fees are sent to the treasury after being settled
+  const totalTreasuryFees = totalDisputesFees.add(treasuryBalance)
```

Comment:
Yes that's correct. Since we are triggering every tx right after the other without waiting for them to be mined we must assume that all of them will succeed and try to withdraw the total amount. 

In the case that one of the settlement tx fails probably withdrawing will also fail. I think it's not too big of a deal since they can try again. 

And when we have implemented the signer experience we will be able to handle this scenarios accordingly.